blueflood-elasticsearch/src/main/java/com/rackspacecloud/blueflood/io/ElasticTokensIO.java

```diff
@@ -45,22 +48,56 @@ public ElasticTokensIO() {
         this.elasticsearchRestHelper = ElasticsearchRestHelper.getInstance();
     }
 
-    @Override
+    public void setElasticsearchRestHelper(ElasticsearchRestHelper elasticsearchRestHelper) {
+        this.elasticsearchRestHelper = elasticsearchRestHelper;
+    }
+
+    @VisibleForTesting
     public void insertDiscovery(Token token) throws IOException {
         List<Token> batch = new ArrayList<>();
         batch.add(token);
         insertDiscovery(batch);
     }
 
-    @Override
-    public void insertDiscovery(List<Token> tokens) throws IOException {
+    /**
+     * A hacky de-generified implementation of two semi-compatible methods from the interfaces implemented here. One
+     * interface expects a list of metrics and the other a list of tokens. This is from an unfortunate copy/paste
+     * incident in the dark past. The {@link com.rackspacecloud.blueflood.inputs.processors.DiscoveryWriter} calls this
+     * to create discovery data for metrics as they're ingested. Originally, it was a bad copy of that class that called
+     * this and passed a list of tokens. Now this lives behind DiscoveryWriter as a discovery module, as appropriate,
+     * and it receives a list of metrics, as it should. Tests still exist that send it a list of tokens. Once those
+     * tests are fixed, this method can change to its correct signature of accepting a list of metrics.
+     */
+    public void insertDiscovery(List things) throws IOException {
+        if (things.get(0) instanceof Token) {
```

Comment:
Existing code had a size() == 0 check. Do we need it to avoid IndexOutOfBoundsException?

Reply:
never mind. Usages of this method always have non-empty list input.

Reply:
Good call. It does work fine right now, but I'm all for guarding against future misuse. [Fixed](https://github.com/rax-maas/blueflood/compare/61529b6eb6fb894c2ff360bb3b628dd73b71edf3..c009776e134915e5417c6642419dd20ea740f894#diff-4e793beb5eafe1d535bc7b3ec8a9bd32a6745f0b51a846e91a38bf0f7401da9bR161-R165).